client/src/main/scala/org/http4s/client/ConnectionManager.scala

```diff
@@ -13,11 +13,15 @@ import scalaz.concurrent.Task
   * must have a mechanism to free resources associated with it.
   */
 trait ConnectionManager[A <: Connection] {
+
+  /** Bundle of the connection and wheither its new or not */
+  final case class NextConnection(connection: A, fresh: Boolean)
```

Comment:
My favorite part is that its a path dependent type and thus is more difficult to share with the wrong `ConnectionManager` (as if anybody ever would accidentally do that...) and it doesn't need type parameters.